.github/workflows/deploy_production.yml

```diff
@@ -34,22 +83,72 @@ jobs:
           restore-keys: |
             ${{ runner.os }}-gradle-
 
-      - name: Decode singing properties
-        run: echo "${{ secrets.ANDROID_SIGNING_PROPERTIES }}" | base64 --decode > signing.properties
+      - name: Install SSH key - For accessing match repo
+        uses: webfactory/ssh-agent@v0.7.0
+        with:
+          ssh-private-key: ${{ secrets.SSH_PRIVATE_KEY }}
+
+      - uses: actions/cache@v3
+        id: bunlderCache
+        with:
+          path: vendor/bundle
+          key: ${{ runner.os }}-gems-${{ hashFiles('**/Gemfile.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-gems-
+
+      - name: Bundle install
+        run: bundle install --path vendor/bundle
+
+      - name: Cache Pods
+        uses: actions/cache@v2
+        id: cocoapodCache
+        with:
+          path: Pods
+          key: ${{ runner.os }}-pods-${{ hashFiles('**/Podfile.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-pods-
 
-      - name: Decode release keystore
-        run: echo "${{ secrets.ANDROID_RELEASE_KEYSTORE }}" | base64 --decode > config/release.keystore
+      - name: Install Pods Dependencies
+        run: bundle exec pod install
+        shell: bash
 
       - name: Decode buildKonfig properties
-        run: echo "${{ secrets.BUILD_KONFIG_PROPERTIES }}" | base64 --decode > buildKonfig.properties
+        run: |
+          cd ..
+          echo ${{ secrets.BUILD_KONFIG_PROPERTIES }} | base64 --decode > buildKonfig.properties
 
-      - name: Build production APK
-        run: ./gradlew assembleProductionRelease -PversionCode=$GITHUB_RUN_NUMBER
+      - name: Paste Google-Info.plist from .secrets
+        run: |
+          mkdir -p sample/Configurations/Plists/GoogleService/Staging
+          echo ${{ secrets.IOS_GOOGLE_INFO_PLIST }} | base64 --decode > ./sample/Configurations/Plists/GoogleService/Staging/GoogleService-Info.plist
 
-      - name: Deploy production to Firebase
-        uses: wzieba/Firebase-Distribution-Github-Action@v1.7.0
-        with:
-          appId: ${{ vars.ANDROID_FIREBASE_APP_ID }}
-          serviceCredentialsFileContent: ${{ secrets.FIREBASE_SERVICE_ACCOUNT_CREDENTIAL_FILE_CONTENT }}
-          groups: nimble
-          file: sample/android/build/outputs/apk/production/release/android-production-release.apk
+      - name: Paste Constants for Fastlane from .secrets
+        run: |
+            mkdir -p fastlane/Constants
+            echo ${{ secrets.IOS_FASTLANE_CONSTANTS }} | base64 --decode > ./fastlane/Constants/Constant.swift
+
+      - name: Generate KMM frameworks for Cocoapods
+        run: |
+          cd ..
+          ./gradlew generateDummyFramework
```

Comment:
@phongvhd93 2 cd steps will move the current dir `./sample/ios` back to the root template `./`. This does not seem to correct 💭 .

Reply:
`./` mean the current folder. If I remove the `gradlew` file in the `sample` folder, this workflow will fail.